From c9cb400d867f2473d56d2a79d39b88634276f164 Mon Sep 17 00:00:00 2001
From: Ayman Grais <ayman.grais@morsemicro.com>
Date: Tue, 28 Apr 2026 06:16:20 +0000
Subject: [PATCH] SW-19863 Increase the SPI inter-block delay to 6us (MM8108)
 (pull request #2103)

In some rare situations, we are hitting SPI reads/writes
failues with a-msdu enabled. This is only hitting when running
a bidirectional TCP traffic with A-MSDU enabled (will also hit
in multista runs if ever needed with SPI). While the block size
remains the same capped at 512 bytes, number of transactions per
packet now increases. While also for larger transfers, the chip
will be busy serving the other hw blocks (MPE).

This seems to be resulting in a higher system bus contention
with incoming data coming to MPE/pktmem, and ongoing data coming
from the host through SPI/YAPS also require bus access. The increased
interblock delay will relax this contention slightly.



Approved-by: Sahand Maleki
Approved-by: Dan Callaghan
Approved-by: Wonseok Kim
Approved-by: Lewis O'Flynn
---
 mm8108.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm8108.c b/mm8108.c
index 61ff247e..bf9db8c3 100644
--- a/mm8108.c
+++ b/mm8108.c
@@ -58,7 +58,7 @@
 #define MM8108_REG_RC_CLK_POWER_OFF_MASK	0x00000040
 #define MM8108_SLOW_RC_POWER_ON_DELAY_MS	2
 
-#define MM8108_SPI_INTER_BLOCK_DELAY_BURST16_NS	4800
+#define MM8108_SPI_INTER_BLOCK_DELAY_BURST16_NS	6000
 #define MM8108_SPI_INTER_BLOCK_DELAY_BURST8_NS	8000
 #define MM8108_SPI_INTER_BLOCK_DELAY_BURST4_NS	15000
 #define MM8108_SPI_INTER_BLOCK_DELAY_BURST2_NS	30000
-- 
2.34.1

